api/src/main/java/eu/darkbot/api/managers/StarSystemAPI.java

```diff
@@ -39,13 +40,41 @@ public interface StarSystemAPI extends API.Singleton {
      */
     Collection<? extends GameMap> getMaps();
 
+    /**
+     * Find {@link GameMap} by given {@code mapId} otherwise will create a new one with given mapId.
+     *
+     * @param mapId id to find
+     * @return {@link GameMap} with given {@code mapId}
+     * @since 0.7.1
+     */
+    GameMap getOrCreateMap(int mapId);
+
+    /**
+     * Find optional {@link GameMap} by given {@code mapId}.
+     *
+     * @param mapId id to find
+     * @return Optional {@link GameMap} with given {@code mapId}, {@link Optional#empty()} otherwise
+     * @since 0.7.1
+     */
+    Optional<GameMap> findMap(int mapId);
+
+    /**
+     * Find optional {@link GameMap} by given {@code mapName}.
+     *
+     * @param mapName name to find
+     * @return Optional {@link GameMap} with given {@code mapName}, {@link Optional#empty()} otherwise
+     * @since 0.7.1
+     */
+    Optional<GameMap> findMap(String mapName);
+
     /**
      * Find {@link GameMap} by given {@code mapId}.
      *
      * @param mapId to find
      * @return {@link GameMap} with given {@code mapId}
      * @throws MapNotFoundException if map was not found
      */
+    @Deprecated(since = "0.7.1")
```

Comment:
```suggestion
    @Deprecated(since = "0.7.0")
```
we have not release 0.7.0 yet, this PR will be part of it